src/services/source/nativeIntegration.ts

```diff
@@ -25,21 +27,56 @@ export default class NativeIntegrationSourceService implements IntegrationSource
     return metaTO;
   }
 
+  private getSourceVersionsMap(): object {
+    const versionSourceMap: object = {};
+    const versions = ["v0", "v1"]
+    versions.forEach(version => {
+      const directoryPath = path.resolve(`src/${version}/sources`);
+      const files = fs.readdirSync(directoryPath, { withFileTypes: true });
+      versionSourceMap[version] = files.filter(file => file.isDirectory())
+        .map(folder => folder.name)
+    });
+    return versionSourceMap;
+  }
+
   public async sourceTransformRoutine(
-    sourceEvents: NonNullable<unknown>[],
+    sourceEvents: unknown[],
     sourceType: string,
     version: string,
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    _requestMetadata: NonNullable<unknown>,
+    _requestMetadata: object,
   ): Promise<SourceTransformationResponse[]> {
-    const sourceHandler = FetchHandler.getSourceHandler(sourceType, version);
-    const respList: SourceTransformationResponse[] = await Promise.all<FixMe>(
+    // if shopify/v1 , webhook/v1 (error) => webhook/v0
+    let sourceVersion = version;
+    // version and sourceEvents structure handling
+    const sourceVersionsMap: object = this.getSourceVersionsMap();
+    if (!sourceVersionsMap[version].includes(sourceType)) {
+      if (version === "v1") {
+        sourceVersion = "v0"
+        // eslint-disable-next-line no-param-reassign
+        sourceEvents = (sourceEvents as SourceInput[]).map(sourceEvent => sourceEvent.event);
+      } else {
+        sourceVersion = "v1"
+        // eslint-disable-next-line no-param-reassign
+        sourceEvents = sourceEvents.map(sourceEvent => ({ event: sourceEvent, source: undefined }));
```

Comment:
Why would source be undefined?

Reply:
This is due to condition that /v0/ is hit for a source in /v1